filament/backend/src/vulkan/VulkanPipelineCache.cpp

```diff
@@ -423,31 +464,48 @@ void VulkanPipelineCache::bindProgramBundle(const ProgramBundle& bundle) noexcep
 }
 
 void VulkanPipelineCache::bindRasterState(const RasterState& rasterState) noexcept {
-    VkPipelineRasterizationStateCreateInfo& raster0 = mPipelineKey.rasterState.rasterization;
-    const VkPipelineRasterizationStateCreateInfo& raster1 = rasterState.rasterization;
-    VkPipelineColorBlendAttachmentState& blend0 = mPipelineKey.rasterState.blending;
-    const VkPipelineColorBlendAttachmentState& blend1 = rasterState.blending;
-    VkPipelineDepthStencilStateCreateInfo& ds0 = mPipelineKey.rasterState.depthStencil;
-    const VkPipelineDepthStencilStateCreateInfo& ds1 = rasterState.depthStencil;
-    VkPipelineMultisampleStateCreateInfo& ms0 = mPipelineKey.rasterState.multisampling;
-    const VkPipelineMultisampleStateCreateInfo& ms1 = rasterState.multisampling;
+    auto& raster0 = mPipelineKey.rasterState.rasterization;
+    const auto& raster1 = rasterState.rasterization;
+    auto& blend0 = mPipelineKey.rasterState.blending;
+    const auto& blend1 = rasterState.blending;
+    auto& ds0 = mPipelineKey.rasterState.depthStencil;
+    const auto& ds1 = rasterState.depthStencil;
+    auto& ms0 = mPipelineKey.rasterState.multisampling;
+    const auto& ms1 = rasterState.multisampling;
     if (
```

Comment:
any sense of if this should be likely or unlikely?

Reply:
Good Q.  Let me add an "unlikely" over the entire expression, that would help with horrible models like car.zip